std-bits/base/src/main/java/org/enso/base/file_format/FileFormatSPI.java

```diff
@@ -0,0 +1,28 @@
+package org.enso.base.file_format;
+
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Value;
+
+import java.util.ServiceLoader;
+
+public abstract class FileFormatSPI {
+  private static final ServiceLoader<FileFormatSPI> loader =
+      ServiceLoader.load(FileFormatSPI.class, FileFormatSPI.class.getClassLoader());
+
+  public static Value[] get_types(boolean refresh) {
+    if (refresh) {
+      loader.reload();
+    }
+    return loader.stream().map(provider -> provider.get().getTypeObject()).toArray(Value[]::new);
+  }
+
+  public Value getTypeObject() {
```

Comment:
Does this method have to be public? Isn't it only used from line 16 from the same file?